### PR TITLE
Liberate OP_RETURN

### PR DESCRIPTION
**Pull request by kyuupichan**
Make multiple OP_RETURN outputs standard,  and place no standardness restrictions on their size

### CODE REVIEW

**Comment by AndyChenW** [comment]
I really really love this PR, once it merged, my code will be much cleaner.

**Comment by Danconnolly** [comment]
Thank you, we'll certainly look at this.

**Comment by corper** [comment]
If we enable nonstandard output:
1. Nonstandard output can hold data also just OP_RETURN. And nonstandard output can hold more data than OP_RETURUN.
2. There will be more than one nonstandard output just like what is going to do in this pull request.
3. Nonstandard output is a real output. Maybe there will be some cooler ideas base on it.

Since OP_RETURN doesn't explode UTXOs. The output satoshi value should can be 0 just like OP_RETURN. We should increase the tx size limit if we want one output can hold more data just like we want to increase OP_RETURN size limit.

In conclusion, I think OP_RETURN is a particular case of nonstandard output. Nonstandard output can do what OP_RETURN do and more.

**Comment by wizenseeker213** [comment]
It is always good to get great information

**Comment by monkeylord** [comment]
> If we enable nonstandard output:
> 
> 1. Nonstandard output can hold data also just OP_RETURN. And nonstandard output can hold more data than OP_RETURUN.
> 2. There will be more than one nonstandard output just like what is going to do in this pull request.
> 3. Nonstandard output is a real output. Maybe there will be some cooler ideas base on it.
> 
> Since OP_RETURN doesn't explode UTXOs. The output satoshi value should can be 0 just like OP_RETURN. We should increase the tx size limit if we want one output can hold more data just like we want to increase OP_RETURN size limit.
> 
> In conclusion, I think OP_RETURN is a particular case of nonstandard output. Nonstandard output can do what OP_RETURN do and more.

I agreed.
We can use nonstandard output like this:
OP_DUP OP_HASH160 <SigHash> OP_EQUALVERIFY OP_CHECKSIG <Data>  OP_DROP <Data> OP_DROP ...

Then we can store whole file in single TX.
File protocol will be like this.
Output 0: OP_RETURN <MetaInfo of File>
Output 1-X: P2SH <FileData>

Since transaction fees is paid after all, storing file in nonstandard output is not only much more efficient, but also easy to decode.
We can build a lot of things based on file, such as serverless website.

**Comment by Danconnolly** [comment]
We've pulled this into our internal repository as a pull request and are analysing the impact.

**Comment by dymurray** [comment]
Any update on whether this will be included in the next release or not?

**Comment by corper** [comment]
I change my mind. I support this PR. It's a small step toward to enable nonstandard output.

**Comment by kyuupichan** [comment]
@corper indeed - there are a few rules about non-standardness.  Remove them one by one, safely.  For current usecases the OP_RETURN restrictions are the most troublesome IMO.  The rest (dust, etc.) can continue to be done incrementally.

**Comment by AndyChenW** [comment]
@Danconnolly is there any update on this? bitcoin sv code is not updated since last Dec on github.

**Comment by mpapec** [comment]
https://whatsonchain.com/tx/ef21e71d00b9fce174222e679640b09e29ac8a55f321c93e64b16cc3109959f8 👍

**Comment by screeni** [comment]
Done Op-Return Bitcoin SV >>> https://whatsonchain.com/stats and https://coingeek.com/bitcoin-sv-op_return-stats-unlocked-by-whatsonchain/

**Comment by Danconnolly** [comment]
OP_RETURN will be fully restored in the Genesis Protocol Upgrade.
